Built-in Math Functions (#101)
NuevoFoundation/workshops: content/english/python-basics/Built-in Math Functions/_index.md

---
title: "Built-in Math Functions"
date: 2019-07-28T11:45:38-07:00
draft: false
weight: 9
---

Python provides a number of important built-in functions that we can use without needing to provide the function definition. In this section, we will learn about some of the built-in math functions using that you can perform mathematical tasks on numbers.

> The  `max()` and  `min()` functions give us the largest and smallest values in a list, respectively:

Reviewer: ```suggestion
The `max()` and  `min()` functions give us the largest and smallest values in a list, respectively:
```

```python

> x = min(20, 10, 50, 25)		#finds the minimum of 20, 10, 50 and 25

Reviewer: ```suggestion
x = min(20, 10, 50, 25)		#x = 10
```

print(x)

> y = max(20, 10, 50, 25)		#finds the maximum of 20, 10, 50 and 25

Reviewer: ```suggestion
y = max(20, 10, 50, 25)		#y = 50
```

print(y)
```

> The  `abs()` function give us the absolute(positive) value of number:

Reviewer: nit: remove extra space between The and `

Reviewer: ```suggestion
The `abs()` function give us the absolute(positive) value of number:
```

```python

> x = abs(-34)		#x = 34

Reviewer: Can you line these comments up?

Reviewer: Use spaces instead of tabs.

print(x)

> y = abs(90)			         #x = 90

Reviewer: ```suggestion
y = abs(90)			         #y = 90
```

print(y)
```

> The  `pow(a, b)` function give us the value of a raised to the power of b (a<sup>b</sup>):

Reviewer: nit: remove extra space between The and `

Reviewer: ```suggestion
The `pow(a, b)` function give us the value of `a` raised to the power of `b` (a<sup>b</sup>):
```

```python

> x = pow(3,4)		        #x  = 81

Reviewer: ```suggestion
x = pow(3,4)		        #x = 81
```

print(x)
```

### Challenge

> Let’s put it all together! Let’s see if we can create a program that takes 5 numbers from the user (say 2,7,4,11,9) and print the value of minimum of those numbers raised to the power of maximum of them (2<sup>11</sup>).

Reviewer: ```suggestion
Let’s put it all together! Let’s see if we can create a program that takes 5 numbers from the user and prints the value of minimum of those numbers raised to the power of maximum. For example, given the number 2,7,4,11,9, return 2<sup>11</sup>.
```
